lib/trento.pm

```diff
@@ -1039,26 +1039,38 @@ sub cypress_test_exec {
     my ($cypress_test_dir, $test_tag, $timeout) = @_;
     $timeout //= bmwqemu::scale_timeout(600);
     my $ret = 0;
-
-    my $test_file_list = script_output("find $cypress_test_dir/cypress/integration/$test_tag -type f -iname \"*.js\"");
+    my $cy_test_struct = 'cypress/integration';
+    $cy_test_struct = 'cypress/e2e'
+      if ((get_var('TRENTO_CYPRESS_VERSION', CYPRESS_DEFAULT_VERSION) =~ /(\d+)\..*/g) &&
+        (int($1) >= 10));
+    my $find_cmd = join(' ',
+        'find',
+        "$cypress_test_dir/$cy_test_struct/$test_tag",
+        '-type', 'f',
+        '-iname', '"*.js"');
+    my $test_file_list = script_output($find_cmd);
 
     for (split(/\n/, $test_file_list)) {
         # Compose the JUnit .xml file name, starting from the .js filename
         my $test_filename = basename($_);
         my $test_result = 'test_result_' . $test_tag . '_' . $test_filename;
         $test_result =~ s/js$/xml/;
-        my $test_cmd = 'run' .
-          ' --spec \"cypress/integration/' . $test_tag . '/' . $test_filename . '\"' .
-          ' --reporter junit' .
-          ' --reporter-options \"mochaFile=/results/' . $test_result . ',toConsole=true\"';
+        my $test_cmd = join(' ', 'run',
```

Comment:
I suspect that this part is somehow not fine. For example in http://openqaworker15.qa.suse.cz/tests/130994

In particular http://openqaworker15.qa.suse.cz/tests/130994#step/test_trento_web/77

```
Cypress exec:run --spec "cypress/e2e/first_visit/test_eula.cy.js" --reporter junit --reporter-options "mochaFile=/results/test_result_first_visit_test_eula.cy.xml,toConsole=true"
```

and the resulting full command is http://openqaworker15.qa.suse.cz/tests/130994#step/test_trento_web/85

```
podman run -it --name goofy -v /root/result:/results -v /root/test/test:/e2e -w /e2e -e "DEBUG=cypress:*" --entrypoint='["/bin/sh", "-c", "/usr/local/bin/cypress run --spec "cypress/e2e/first_visit/test_eula.cy.js" --reporter junit --reporter-options "mochaFile=/results/test_result_first_visit_test_eula.cy.xml,toConsole=true" 2>/results/cypress_first_visit_log.txt"]' docker.io/cypress/included:12.1.0 | tee cypress_first_visit_result.txt; echo rHKmY-$?-

```

It result in error like

```
Error: runc create failed: unable to start container process: exec: "[\"/bin/sh\", \"-c\", \"/usr/local/bin/cypress run --spec \"cypress/e2e/first_visit/test_eula.cy.js\" --reporter junit --reporter-options \"mochaFile=/results/test_result_first_visit_test_eula.cy.xml,toConsole=true\" 2>/results/cypress_first_visit_log.txt\"]": stat ["/bin/sh", "-c", "/usr/local/bin/cypress run --spec "cypress/e2e/first_visit/test_eula.cy.js" --reporter junit --reporter-options "mochaFile=/results/test_result_first_visit_test_eula.cy.xml,toConsole=true" 2>/results/cypress_first_visit_log.txt"]: no such file or directory: OCI runtime attempted to invoke a command that was not found

```

Reply:
As reference an execution with test code from the top of master branch

http://openqaworker15.qa.suse.cz/tests/130967#step/test_trento_web/116

```
run --spec \"cypress/integration/first_visit/test_eula.js\" --reporter junit --reporter-options \"mochaFile=/results/test_result_first_visit_test_eula.xml,toConsole=true\"
```

Reply:
Yes, there's something odd here.

I did this:
```
# cat t.pl 
$cy_test_struct = 'cypress/integration';
$test_tag = 'first_visit';
$test_filename = 'test_eula.js';
$test_result = 'test_result_first_visit_test_eula.xml';

 my $test_cmd = join(' ', 'run',
            '--spec', '\"' . $cy_test_struct . '/' . $test_tag . '/' . $test_filename . '\"',
            '--reporter', 'junit',
            '--reporter-options', '\"mochaFile=/results/' . $test_result . ',toConsole=true\"');

print "CMD: [$test_cmd]\n";

# perl t.pl
CMD: [run --spec \"cypress/integration/first_visit/test_eula.js\" --reporter junit --reporter-options \"mochaFile=/results/test_result_first_visit_test_eula.xml,toConsole=true\"]
```
And I got the expected result.